src/components/AutoComplete/AutoComplete.tsx

```diff
@@ -0,0 +1,196 @@
+import useTheme from "../../hooks/useTheme";
+import Icon from "../Icon";
+import Field from "../internal/Field";
+import InternalInput from "../internal/InternalInput";
+import LoadingIcon from "../LoadingIcon";
+import VisuallyHidden from "../VisuallyHidden";
+import { defaultAutoCompleteProps } from "./defaultAutoCompleteProps";
+import { AutoCompleteProps, ListItemKey } from "./types";
+import useGetItems from "./useGetItems";
+
+const getFieldErrors = (
+  error: string | string[] | undefined
+): { fieldErrors: string[] | undefined; hasErrors: boolean } => {
+  if (error === undefined) {
+    return { fieldErrors: undefined, hasErrors: false };
+  }
+
+  if (Array.isArray(error)) {
+    return {
+      fieldErrors: error,
+      hasErrors: true,
+    };
+  }
+
+  return { fieldErrors: [error], hasErrors: true };
+};
+
+function AutoComplete<Item extends ListItemKey = ListItemKey>(
+  props: AutoCompleteProps<Item>
+) {
+  const mergedProps = useMergedProps(props, defaultAutoCompleteProps);
+  const theme = useTheme();
+
+  const {
+    label,
+    innerRef,
+    error,
+    onChange,
+    onBlur,
+    onFocus,
+    disabled,
+    helpText,
+    itemToString: itemToStringFn,
+    placeholder,
+    itemsFooter: Footer,
+    listItem: ListItem,
+    hideLabel,
+    testId,
+    value,
+    optional,
+    __internal__open,
+    __internal__highlightedIndex,
+    __internal__loading,
+    __internal__focus,
+  } = mergedProps;
+
+  const { items, getItems, status } = useGetItems(props.getItems);
+
+  const auxId = useMemo(() => `auto-complete-aux-${nanoid()}`, []);
+  const { fieldErrors, hasErrors } = getFieldErrors(error);
+  const describedBy = helpText || hasErrors ? auxId : undefined;
+
+  const itemToString = (item: Item | null): string =>
+    itemToStringFn ? itemToStringFn?.(item) : item ? String(item) : "";
+
+  const {
+    isOpen,
+    getMenuProps,
+    getInputProps,
+    getItemProps,
+    getComboboxProps,
+    getToggleButtonProps,
+    highlightedIndex,
+    inputValue,
+    selectItem,
+    closeMenu,
+  } = useCombobox<Item | null>({
+    items: items || [],
+    defaultSelectedItem: value,
+    onInputValueChange: (changed) => {
+      getItems(changed);
+    },
+    onSelectedItemChange: (changed) => {
+      onChange?.(changed.selectedItem);
+    },
+    itemToString: (item) =>
+      itemToStringFn ? itemToStringFn?.(item) : item ? String(item) : "",
+  });
+
+  const onClear = () => {
+    selectItem(null);
+    closeMenu();
+  };
+
+  const menuIsOpen = isOpen || (__internal__open ?? false);
+  const showClearIcon = (menuIsOpen && items.length > 0) || inputValue !== "";
+  const showMagnifier = !showClearIcon;
+  const loading = status === "LOADING" || __internal__loading;
+
+  const renderListItem = (record: Item) => {
+    if (ListItem) {
+      return <ListItem inputValue={inputValue} item={record} />;
+    }
+    return itemToString ? itemToString(record) : record;
```

Comment:
so if getItems is simple static list, then we should handle `record` as an object. 
in otherwords 
```
const getItmes = ()=> ["Item1", "Item2"]
```
User won't pass itemToString to us since it is optional . Therefore this bit of code will only return `record`